pkg/applications/test/kubernetes.go

```diff
@@ -65,6 +66,60 @@ func CreateNamespaceWithCleanup(t *testing.T, ctx context.Context, client ctrlru
 	return ns
 }
 
+// EnsureClusterWithCleanup creates a cluster resource with a static name "default" and needed status fields for fetching it
+// based on the namespacedname and registers a hook on T.Cleanup that removes it at the end of the test.
+func EnsureClusterWithCleanup(t *testing.T, ctx context.Context, client ctrlruntimeclient.Client) {
+	cluster := &kubermaticv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "default",
+		},
+		Spec: kubermaticv1.ClusterSpec{
+			HumanReadableName: "humanreadable-cluster-name",
+			ClusterNetwork: kubermaticv1.ClusterNetworkingConfig{
+				ProxyMode: "iptables",
+				Pods: kubermaticv1.NetworkRanges{
+					CIDRBlocks: []string{"10.0.0.0/24"},
+				},
+				Services: kubermaticv1.NetworkRanges{
+					CIDRBlocks: []string{"10.1.0.0/24"},
+				},
+			},
+			ExposeStrategy: "NodePort",
+		},
+	}
+	if err := client.Create(ctx, cluster); err != nil {
+		t.Fatalf("failed to create test cluster: %s", err)
```

Comment:
Please settle on `%v` for the placeholder, instead of switching between %v and %s.